code/datums/mutations.dm

```diff
@@ -142,18 +145,28 @@
 				apply_overlay(CM.layer_used)
 
 /datum/mutation/human/proc/modify() //called when a genome is applied so we can properly update some stats without having to remove and reapply the mutation from someone
-	if(modified || !power || !owner)
+	if(!owner)
 		return
-	power.charge_max *= GET_MUTATION_ENERGY(src)
-	power.charge_counter *= GET_MUTATION_ENERGY(src)
-	modified = TRUE
+	power_dormant = !GET_MUTATION_SYNCHRONIZER(src) //Granting dormant powers is only by synchronizer chromosome so far
+	if(power)
+		var/power_given = FALSE
+		for(var/spell in owner.mind.spell_list)
+			if(spell == power)
+				power_given = TRUE
+		if(power_given && power_dormant) //Power going back into dormancy
+			owner.RemoveSpell(power)
+		if(!power_given && !power_dormant)
+			grant_spell()
+		if(istype(power))
```

Comment:
how can this check fail?

Reply:
I'm basically using this as an isnull here, since removing the power from it going dormant would cause null refs in changing the power variables. It istype an acceptable way to do this?